### PR TITLE
[WIP] Remove Swift 3 @objc inference support.

### PR DESCRIPTION
**Pull request by rudkx**
This does not include any test updates. I will open a separate PR for them and merge that PR before this one.

### CODE REVIEW

File: lib/Sema/TypeCheckDeclOverride.cpp [hunk 1]
```diff
@@ -1203,39 +1203,7 @@ namespace  {
...
-      Diags.diagnose(Override, diag::override_swift3_objc_inference,
-                     Override->getDescriptiveKind(),
-                     Override->getFullName(),
-                     Base->getDeclContext()
-                       ->getAsNominalTypeOrNominalTypeExtensionContext()
-                       ->getName());
-      Diags.diagnose(Base, diag::make_decl_objc, Base->getDescriptiveKind())
-        .fixItInsert(Base->getAttributeInsertionLoc(false),
-                     "@objc ");
+      // FIXME: Removing this results in compile-time errors.
```
Comment (rudkx) [inline lib/Sema/TypeCheckDeclOverride.cpp]: @DougGregor I'm apparently missing what meta-programming magic is requiring an implementation of visitObjCAttr here.

Comment (DougGregor) [reply lib/Sema/TypeCheckDeclOverride.cpp]: You can use `UNINTERESTING_ATTR(ObjC)` in place of an empty function body.

**Comment by rudkx** [comment]
@swift-ci Please smoke test

**Comment by rudkx** [comment]
This has a bunch of conflicts at this point; closing.

**Pull request closed without merging**